apps/sms/test/unit/thread_ui_test.js

```diff
@@ -2208,43 +2253,67 @@ suite('thread_ui.js >', function() {
         ThreadUI.renderMessages(1);
       });
 
-      test('infinite rendering test', function() {
+      test('infinite rendering test', function(done) {
         var chunkSize = ThreadUI.CHUNK_SIZE;
         var message;
 
-        for (var i = 1; i < chunkSize; i++) {
-          MessageManager.getMessages.yieldTo(
-            'each', MockMessages.sms({ id: i })
-          );
-          message = document.getElementById('message-' + i);
-          assert.ok(
-            message.classList.contains('hidden'),
-            'message-' + i + ' should be hidden'
-          );
+        function assertHidden() {
+          return new Promise((resolve, reject) => {
```

Comment:
Sorry, don't really get why we need ```new Promise()``` here and in ```assertFirstChunk```? We don't call nor resolve neither reject so these promises are never resolved (rejected only if something fails, but the same would be true without promise wrapper).

Can't we just do ```TaskRunner.prototype.push.lastCall.returnValue.then(assertHidden).then(done, done);``` and add ```return``` for the last statement in ```assertHidden``` and ```assertFirstChunk```?